### DISCUSSION TITLE
gatsby-source-filesystem not sourcing files from site directory when used in a theme

### DISCUSSION THREAD

**Discussion by AlexanderProd** [gatsbyjs/gatsby#36103] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I'm trying to build a gatsby theme that uses `gatsby-source-filesystem` to source images.

I have set up a yarn workspace for development, my folder structure looks like this.

```
workspace/
├─ example.com/
│  ├─ src/
│  │  ├─ pages/
│  │  │  ├─ test.js
│  ├─ gatsby-config.js
│  ├─ package.json
├─ gatsby-theme-example/
│  ├─ src/
│  ├─ images/
│  ├─ gatsby-config.js
│  ├─ gatsby-node.js
│  ├─ package.json
```

The yarn workspace is also set up correctly and the `gatsby-theme-example` is a dependency of `example.com`, `yarn workspaces info` correctly shows it in `workspaceDependencies`. 

Both workspaces have a `gatsby-config.js` file, the `gatsby-theme-example` has the `gatsby-source-filesystem` in it. 

```JS
{
   resolve: `gatsby-source-filesystem`,
   options: {
      name: `images`,
      path: path.join(__dirname, "images"),
   },
},
```

The problem I'm facing is that I have to put the images inside the theme directories image folder, otherwise they are not found by `gatsby-source-filesystem`. From following [this][1] video and reading the [tutorial][2], using `path.join` and the `__dirname` variable should point to the package that has the theme installed as a dependency, in my case `example.com`

Strangely, the `gatsby-plugin-page-creator` plugin in `gatsby-theme-example/gatsby-config.js` creates the pages defined in `example.com/src/pages` (the site directory) perfectly.

```JS
{
   resolve: "gatsby-plugin-page-creator",
   options: {
      path: path.join(__dirname, "src/pages"),
   },
},
```

I also have `onPreBootstrap` defined in `gatsby-theme-example/gatsby-node.js` like this 

```JS
exports.onPreBootstrap = ({ reporter }) => {
  const contentPath = `${__dirname}/images/`
  if (!fs.existsSync(contentPath)) {
    reporter.info(`creating the ${contentPath} directory`)
    fs.mkdirSync(contentPath)
  }
}
```

and it creates the images directory in `gatsby-theme-example` not `example.com`

I'm using `gatsby@4.18.0` and `gatsby-source-filesystem@4.18.0` and start the project using this command: `yarn workspace example.com start`

  [1]: https://www.youtube.com/watch?v=NkW06HK9-aY&t=161s
  [2]: https://www.gatsbyjs.com/tutorial/building-a-theme/#add-static-data-to-a-theme

### Reproduction Link

https://github.com/AlexanderProd/gatsby-source-filesystem-theme-bug

### Steps to Reproduce

1. clone the repository
2. `yarn workspace example.com install`
3. `yarn workspace example.com start`
4.  go to http://localhost:8000/___graphql
5. run the following query 

```
query MyQuery {
  allFile {
    edges {
      node {
        id
        name
      }
    }
  }
}

```

### Expected Result

Expected the files in `example.com/images/` to be shown in the graphql query. 

### Actual Result

No files are found, even though `gatsby-source-filesystem` is configured and files are present in the `example.com/images/` directory.

### Environment

```shell
System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /var/folders/sg/7qlzh78n07d0n3v6bzh_twch0000gn/T/yarn--1657532819124-0.8786376693783073/node
    Yarn: 1.22.18 - /var/folders/sg/7qlzh78n07d0n3v6bzh_twch0000gn/T/yarn--1657532819124-0.8786376693783073/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Chrome: 99.0.4844.83
    Firefox: 99.0.1
    Safari: 15.5
  npmGlobalPackages:
    gatsby-cli: 4.13.0
```

**Comment by LekoArts** [comment, marked as answer]
It'll point at the directory inside the theme, not the example site. Instead of `path.join(__dirname, 'images')` try using `images`.

**Reply by AlexanderProd** [reply]
Thanks, gonna try that. 

But why is the path in `gatsby-plugin-page-creator` pointing at the directory in the example site when using `__dirname`?

**Reply by LekoArts** [reply]
Gatsby includes the `gatsby-plugin-page-creator` pointing at `src/pages` by default, so you should have three pages actually, two from your example site, one from the theme.

**Reply by AlexanderProd** [reply]
Yeah I have three pages indeed.

**Reply by AlexanderProd** [reply]
it does work when using just `images` as path.
